Sources/StreamChatUI/ChatMessageList/ChatMessageListCollectionView.swift

```diff
@@ -5,13 +5,109 @@
+    /// its height can change thanks to dynamic type so we need to track its changes
+    private var topContentInset: CGFloat = 0 {
+        didSet { contentInset.top += topContentInset - oldValue }
+    }
+    
+    override open func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        
+        guard !isInitialized, superview != nil else { return }
+        
+        isInitialized = true
+        
+        setUp()
+        setUpLayout()
+        setUpAppearance()
+        updateContent()
+    }
+    
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+        topContentInset = dateView.frame.height + 2 * layoutMargins.top
+    }
+    
+    open func setUp() {
+        // Setup `contentOffset` observation so `delegate` is free for anyone that wants to use it
+        contentOffsetObservation = observe(\.contentOffset) { cv, _ in
+            /// To display correct date we use bottom edge of `dateView` (we use `cv.layoutMargins.top` for both vertical offsets of `dateView`
+            let dateViewRefPoint = CGPoint(x: cv.dateView.center.x, y: cv.dateView.frame.maxY + cv.layoutMargins.top)
+            let refPoint = cv.convert(dateViewRefPoint, to: cv)
+            // To get correct indexPath, we cannot use `cv.indexPathForItem(at:)` as it breaks collectionView layout
+            // and all cells have fix 72pt width, so we search which visible cell contains our `refPoint`
+            // and then search for its indexPath
+            /// Cell that containe
+            let cell = cv.visibleCells.first(where: { $0.frame.contains(refPoint) })
+
+            // If we cannot find any indexPath for `cell` we try to use max visible indexPath (we have bottom to top) layout
+            guard let indexPath = cell.flatMap(cv.indexPath) ?? cv.indexPathsForVisibleItems.max() else { return }
+            
+            let date = cv.chatDelegate?.collectionView(cv, dateForItemAt: indexPath)
+            
+            // As cells can overlay our `dateView` we need to keep it above them
+            cv.bringSubviewToFront(cv.dateView)
+            
+            // If we have no date we have no reason to display `dateView`
+            cv.dateView.isHidden = date == nil
+            cv.dateView.date = date
+        }
+    }
+    
+    open func setUpLayout() {
+        addSubview(dateView)
+        
+        topContentInset = dateView.frame.height + 2 * layoutMargins.top
+        
+        NSLayoutConstraint.activate([
+            dateView.topAnchor.pin(equalTo: layoutMarginsGuide.topAnchor),
+            dateView.centerXAnchor.pin(equalTo: layoutMarginsGuide.centerXAnchor),
+            dateView.leadingAnchor.pin(greaterThanOrEqualTo: layoutMarginsGuide.leadingAnchor),
+            dateView.trailingAnchor.pin(lessThanOrEqualTo: layoutMarginsGuide.trailingAnchor)
+        ])
+    }
+    
+    open func setUpAppearance() {
+        topContentInset = dateView.frame.height + 2 * layoutMargins.top
```

Comment:
nit: DRY for line 62, is this even needed when it gets called on `layoutSubviews()` ?

Reply:
Nope 🙂